language/move-prover/src/prelude.bpl

```diff
@@ -827,12 +827,138 @@ procedure {:inline 1} $Sub(src1: $Value, src2: $Value) returns (dst: $Value)
 // to src2 == any integer Value from 0..127.
 // Left them out for brevity
 function $power_of_2(power: $Value): int {
-    (var p := i#$Integer(power);
-     if p == 32 then 4294967296
-     else if p == 64 then 18446744073709551616
-     // Value is undefined, otherwise.
-     else -1
-     )
+   (var p := i#$Integer(power);
```

Comment:
This looks super dangerous for non-termination problems because this can lead to exponential explosion via case splitting of a very high degree. We can try this out for now but I'm pretty sure we run into issues.

Reply:
Yeah, it has nothing to do with the Role refactoring, and I'll just take it out. I had some problem I don't understand where I suddenly started getting errors from SlidingNonce, and I was trying to fix them.  I agree that it's dangerous.  It may be the only way to implement variable shift, but maybe we need to find another solution to SlidingNonce so we don't need it.